xarray/tests/test_backends.py

```diff
@@ -1267,6 +1267,121 @@ def test_4_open_large_num_files_h5netcdf(self):
+                      coords={
+                          't': (['t', ], t1),
+                          'x': (['x', ], x)
+                      })
+
+        ds2 = Dataset(data_vars={self.var_name: (['t', 'x'], v2),
+                                 self.coord_name: ('x', 2 * x)},
+                      coords={
+                          't': (['t', ], t2),
+                          'x': (['x', ], x)
+                      })
+
+        return ds1, ds2
+
+    def test_open_mfdataset_does_same_as_concat(self):
+        with create_tmp_file() as tmpfile1:
+            with create_tmp_file() as tmpfile2:
+                ds1, ds2 = self.gen_datasets_with_common_coord_and_time()
+
+                # save data to the temporary files
+                ds1.to_netcdf(tmpfile1)
+                ds2.to_netcdf(tmpfile2)
+
+                files = [tmpfile1, tmpfile2]
+
+                options = ['all', 'minimal', 'different', ]
+                for opt in options:
+                    with open_mfdataset(files, data_vars=opt) as ds:
+                        kwargs = dict(data_vars=opt, dim='t')
+                        ds_expect = xr.concat([ds1, ds2], **kwargs)
+                        self.assertDatasetIdentical(ds, ds_expect)
+
+                    with open_mfdataset(files, coords=opt) as ds:
+                        kwargs = dict(coords=opt, dim='t')
+                        ds_expect = xr.concat([ds1, ds2], **kwargs)
+                        self.assertDatasetIdentical(ds, ds_expect)
+
+    def test_common_coord_when_datavars_passed(self):
+        with create_tmp_file() as tmpfile1:
+            with create_tmp_file() as tmpfile2:
+                ds1, ds2 = self.gen_datasets_with_common_coord_and_time()
+
+                # save data to the temporary files
+                ds1.to_netcdf(tmpfile1)
+                ds2.to_netcdf(tmpfile2)
+
+                files = [tmpfile1, tmpfile2]
+
+                for opt in ['all', 'minimal']:
+                    # open the files with the default data_vars='all'
+                    with open_mfdataset(files, data_vars=opt) as ds:
+
+                        coord_shape = ds[self.coord_name].shape
+                        coord_shape1 = ds1[self.coord_name].shape
+                        coord_shape2 = ds2[self.coord_name].shape
+
+                        var_shape = ds[self.var_name].shape
+
+                        tests = []
+                        # shape pairs to be compared
+                        shape_pairs = [
+                            (var_shape, coord_shape),
+                            (coord_shape1, coord_shape),
+                            (coord_shape2, coord_shape)
+                        ]
+                        # tests to be applied to respective pairs
+                        if opt == 'all':
+                            tests = [self.assertEqual,
+                                     self.assertNotEqual, self.assertNotEqual]
```

Comment:
nit: it's best to avoid putting complex control flow in tests -- it makes them harder to debug. I would actually prefer if you wrote two different tests here with a bit more copied code.

Reply:
I've split the test into 2, let me know if this is what you meant.

Cheers